File: src/org/cappuccino_project/ide/intellij/plugin/psi/interfaces/ObjJIsOfClassType.java
package org.cappuccino_project.ide.intellij.plugin.psi.interfaces;

import org.cappuccino_project.ide.intellij.plugin.psi.types.ObjJClassType;
import org.jetbrains.annotations.NotNull;


public interface ObjJIsOfClassType {
    ObjJIsOfClassType UNDEF = new ObjJIsOfClassType() {
        @NotNull
        @Override
        public ObjJClassType getClassType() {
            return ObjJClassType.UNDEF;
        }
    };

    @NotNull
    ObjJClassType getClassType();
}

File: src/org/cappuccino_project/ide/intellij/plugin/psi/interfaces/ObjJIsOfClassType.kt
package org.cappuccino_project.ide.intellij.plugin.psi.interfaces

import org.cappuccino_project.ide.intellij.plugin.psi.types.ObjJClassType


interface ObjJIsOfClassType {

    val classType: ObjJClassType

    companion object {
        val UNDEF: ObjJIsOfClassType = object : ObjJIsOfClassType {
            override val classType: ObjJClassType
                get() = ObjJClassType.UNDEF
        }
    }
}
